### PR TITLE
storage/pg: replication wal sender timeout

### PR DESCRIPTION
**Pull request by sploiselle**
A simpler and more immediate fix to #26813 is to configure the `wal_sender_timeout` on the PG replication client.

For testing, I have checked in dev builds that the PG server respects the setting for the session. In customer environments, if this fails to work, we'll log the error through Sentry. If for, whatever reason, this turns out to be noisy we can reconsider.

Once this merges, I'll add the flag to Launch Darkly.

### Motivation

This PR fixes a recognized bug. Provides a workaround for the shortcoming identified in #26813 

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/adapter/src/flags.rs [hunk 1]
```diff
@@ -43,6 +43,7 @@ pub fn storage_config(config: &SystemVars) -> StorageParameters {
             tcp_user_timeout: Some(config.pg_source_tcp_user_timeout()),
         },
         pg_source_snapshot_statement_timeout: config.pg_source_snapshot_statement_timeout(),
+        pg_source_wal_sender_timeout: config.pg_source_wal_sender_timeout(),
```
Comment (rjobanp) [inline src/adapter/src/flags.rs:46]: setting flags this way has been deprecated -- would switch this to use `dyncfg` instead -- also a lot less boilerplate!

Comment (sploiselle) [reply src/adapter/src/flags.rs:46]: If this has been deprecated, why are all of these options still set this way? I also didn't see an announcement that these code paths have been deprecated––can you link me to it?

Comment (rjobanp) [reply src/adapter/src/flags.rs:46]: https://materializeinc.slack.com/archives/C01CFKM1QRF/p1710780368452619 and https://materializeinc.slack.com/archives/CUXUBJH7B/p1710192018978979

We want to move these over at some point -- we just haven't done so yet, but we shouldn't keep adding new ones if we can avoid it

Comment (rjobanp) [reply src/adapter/src/flags.rs:46]: I guess given that this is setting a timeout closely related to the other pg source timeout settings it's fine to leave as-is and we should migrate them all over at some point

File: src/postgres-util/src/tunnel.rs [hunk 2]
```diff
@@ -94,6 +94,10 @@ impl Default for TcpTimeoutConfig {
 }
 
 pub const DEFAULT_SNAPSHOT_STATEMENT_TIMEOUT: Duration = Duration::ZERO;
+// In PG16, this value defaults to 60 seconds, so we want to esnure we get at
```
Comment (rjobanp) [inline src/postgres-util/src/tunnel.rs:97]:
```suggestion
// In PG16, this value defaults to 60 seconds, so we want to ensure we get at
```

File: src/adapter/src/flags.rs (hunk 1 above)
Comment (sploiselle) [reply src/adapter/src/flags.rs:46]: Nice––I hadn't seen Gus's announcement. None of those spots mention "deprecation," though. Aslo, this PR is green and the work to move this option alongside all of the others is nominal, so don't see the value in re-running CI for this. Next time I need to add an option, though, will plumb it through `dyncfg`.

Comment (sploiselle) [reply src/adapter/src/flags.rs:46]: > I guess given that this is setting a timeout closely related to the other pg source timeout settings it's fine to leave as-is and we should migrate them all over at some point

agreed!

**Review by rjobanp — APPROVED** [review]
one other nit but otherwise LGTM

File: src/postgres-util/src/tunnel.rs (hunk 2 above)
Comment (sploiselle) [reply src/postgres-util/src/tunnel.rs:97]: ugh––going to live with this because it doesn't seem ambiguous enough to run back through CI.

**Pull request merged**